demo/api/src/main.ts

```diff
@@ -41,6 +41,11 @@ async function bootstrap(): Promise<void> {
     app.use(compression());
     app.use(cookieParser());
 
+    // if cdn is enabled, make sure all traffic is coming from the cdn
+    if (config.cdn.enabled) {
+        app.useGlobalGuards(new BlockWithoutHeaderGuard({ headerName: "x-cdn-origin-check", headerValue: config.cdn.originCheck }));
+    }
```

Comment:
How would this work for cluster-internal requests that bypass the CDN? For instance, the site querying during build?

Reply:
Nice catch. We could either attach the header during build (ugly), use a second header (also ugly) or detect that the request is coming from an internal source.

Reply:
@nsams what do you think?

Reply:
I've added a check. Basically, if `x-forwarded-for` is present, we assume the request is coming from outside. This is safe enough, if we trust the reverse proxy.